crates/bevy_app/src/app.rs

```diff
@@ -11,15 +15,22 @@ use bevy_ecs::{
     schedule::{ScheduleBuildSettings, ScheduleLabel},
     system::{IntoObserverSystem, SystemId, SystemInput},
 };
-#[cfg(feature = "trace")]
-use bevy_utils::tracing::info_span;
-use bevy_utils::{tracing::debug, HashMap};
+use bevy_utils::HashMap;
 use core::{fmt::Debug, num::NonZero, panic::AssertUnwindSafe};
+use log::debug;
+use thiserror::Error;
+
+#[cfg(feature = "trace")]
+use tracing::info_span;
```

Comment:
As with `bevy_ecs`, I'm flattening our relationship with `bevy_utils::tracing` and also relying on `log` for non-span logging. It's already been mentioned that we should probably move things like `log` and `tracing` into workspace dependencies, but I'm just maintaining status quo for now.